feat(docker): add mailhog container

hdahlheim opened the pull request:
Adds [MailHog](https://github.com/mailhog/MailHog) to the docker compose for local transactional email testing with the node SMTP mailer.

To make the SMTP mailer work add the following vars to your .env

```
SEND_MAILS=true
SEND_MAILS_TAGS=dev
SEND_MAILS_REGEX_FILTERS=^([a-zA-Z0-9\._%+-]+)@republik\.(ch|test)$ // to allow *@republik.ch or *republik.test
SEND_MAILS_NODEMAILER_TEMPLATE_REGEX='^.+'
SEND_MAILS_NODEMAILER_CONNECTION_URL='smtp://user:password@127.0.0.1:1025'
SEND_MAILS_SUBJECT_PREFIX='Test Local'
```

lucianak reviewed (approved):
LGTM, I haven't tested it though :) I would also add the necessary env-Variables to the test container package readme or to an example.env.test.local file

hdahlheim commented:
@lucianak I moved the `.env` settings into the `apps/api/README.md`. The old `README.md` was extremely out of date, so I deleted most of it. We should invest some time in a different PR to update the document with relevant information.
